src/libcore/str/mod.rs

```diff
@@ -165,6 +165,40 @@ Section: Creating a string
 ///
 /// [`String`]: ../../std/string/struct.String.html#method.from_utf8
 /// [`&str`]: ../../std/str/fn.from_utf8.html
+///
+/// # Examples
+///
+/// This error type’s methods can be used to create functionality
+/// similar to `String::from_utf8_lossy` without allocating heap memory:
+///
+/// ```
+/// fn from_utf8_lossy<F>(mut input: &[u8], mut push: F) where F: FnMut(&str) {
+///     loop {
+///         match ::std::str::from_utf8(input) {
+///             Ok(valid) => {
+///                 push(valid);
+///                 break
+///             }
+///             Err(error) => {
+///                 let (valid, after_valid) = input.split_at(error.valid_up_to());
+///                 unsafe {
+///                     push(::std::str::from_utf8_unchecked(valid))
+///                 }
+///                 push("\u{FFFD}");
+///
+///                 match error.error_len() {
```

Comment:
Just stylistic comment. I think that this `match` taking 8 lines is artificially too long, you could make it 5 lines with `if let`.

```rust
if let Some(invalid_sequence_length) = error.error_len() {
    input = &after_valid[invalid_sequence_length..];
} else {
    break;
}
```

Reply:
This does look better. Applied, thanks.